src/components/frame/FrameConnector.tsx

```diff
@@ -46,50 +36,64 @@ interface LegacyFrameConnectorProps extends BaseFrameConnectorProps {
+  // that way we handle automatically legacy renderer
+  const methods = useMemo<LegacyFrameActions>(() => {
+    return {
+      updateHeight: (height: number) => dispatchProxy(updateHeight(height)),
+      updateTemplates: templates => dispatchProxy(updateTemplates(templates)),
+      handleObfuscation: field => dispatchProxy(obfuscateField(field))
+    };
+  }, [dispatchProxy]);
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore error because of typescript distributivity
-  // FrameConnectorProps can me mapped to UseLegacyChildrenFrameProps which is not correct. It could be solved using
-  // conditional but there is a hooks rules saying shouldn't use conditional
-  const [connected, toFrame] = useChildFrame({ methods, dispatch, iframe });
+  const [connected, toFrame] = useChildFrame({ methods, dispatch: dispatchProxy, iframe });
   useEffect(() => {
     if (connected) {
       onConnected(
-        Object.assign(
-          (action: HostActions) => {
-            if (toFrame.dispatch) {
-              toFrame.dispatch(action);
-            }
-          },
-          {
-            renderDocument: (document: Document, rawDocument?: WrappedDocument<Document>) => {
-              if (toFrame.renderDocument) {
-                toFrame.renderDocument(document, rawDocument);
-              }
-            },
-            selectTemplateTab: (tabIndex: number) => {
-              if (toFrame.selectTemplateTab) {
-                toFrame.selectTemplateTab(tabIndex);
-              }
-            },
-            print: () => {
-              if (toFrame.print) {
-                toFrame.print();
-                return true;
-              }
-              return false;
+        Object.assign((action: HostActions) => {
+          // if toFrame.dispatch is set that means we are on the main track with modern renderer
+          // there is nothing to do but to call dispatch with the action provided.
+          if (toFrame.dispatch) {
+            toFrame.dispatch(action);
+          } else {
+            // otherwise if toFrame.dispatch is NOT set that means that we are dealing with a legacy renderer
+            // in that event we will map each action to it's legacy renderer function
+            // - "RENDER_DOCUMENT" must call toFrame.renderDocument
+            // - "SELECT_TEMPLATE" must call toFrame.selectTemplateTab
+            // - "PRINT" must call toFrame.print
+            if (action.type === "RENDER_DOCUMENT" && toFrame.renderDocument) {
+              toFrame.renderDocument(action.payload.document, action.payload.rawDocument);
+            } else if (action.type === "SELECT_TEMPLATE" && toFrame.selectTemplateTab) {
+              const index = templates.current.findIndex(template => template.id === action.payload);
+              if (index === -1) console.error(`Unable to find the index associated with the label ${action.payload}`);
+              toFrame.selectTemplateTab(index);
```

Comment:
I thought you mentioned there is a case here where the action.payload could be a number also? Are we going to handle that?

Reply:
I got rid of that in the latest commit
- application will use dispatch and fallback to legacy functions
- renderer will use dispatch only
- we will migrate our 2 apps to the latest renderer => no problem